### PR TITLE
QQE-823 | Update image versions

### PR DESCRIPTION
**Pull request by fedinskiy**
### Summary

Part of https://issues.redhat.com/browse/QQE-823
The version update upstream is a part of this PR: https://github.com/quarkusio/quarkus/pull/42078

Please select the relevant options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Dependency update
- [ ] Refactoring
- [ ] Backport
- [ ] New scenario (non-breaking change which adds functionality)
- [ ] This change requires a documentation update
- [x] This change requires execution against OCP (use `run tests` phrase in comment)

### Checklist:
- [x] Methods and classes used in PR scenarios are meaningful
- [x] Commits are well encapsulated and follow [the best practices](https://cbea.ms/git-commit/)

### CODE REVIEW

**Comment by fedinskiy** [comment]
run tests

File: pom.xml [hunk 1]
```diff
@@ -266,15 +266,15 @@
...
+                                <mongodb.image>docker.io/library/mongo:7.0</mongodb.image>
                                 <redis.image>docker.io/library/redis:7.2</redis.image>
                                 <wiremock.image>docker.io/wiremock/wiremock:3.3.1</wiremock.image>
                                 <!-- TODO use official image when/if this fixed https://github.com/hashicorp/docker-consul/issues/184 -->
                                 <consul.image>docker.io/bitnami/consul:1.17.2</consul.image>
                                 <infinispan.image>quay.io/infinispan/server:14.0</infinispan.image>
                                 <infinispan.expected-log>Infinispan Server.*started in</infinispan.expected-log>
                                 <spring.cloud.server.image>quay.io/quarkusqeteam/spring-cloud-config-server:4.1</spring.cloud.server.image>
-                                <keycloak.image>quay.io/keycloak/keycloak:24.0</keycloak.image>
+                                <keycloak.image>quay.io/keycloak/keycloak:25.0</keycloak.image>
```
Comment (jedla97) [inline pom.xml:277]: Keycloak was bumped with https://github.com/quarkus-qe/quarkus-test-suite/pull/1900/files#diff-9c5fb3d1b7e3b0f54bc5c4182965c4fe1f9023d449017cece3005d3f90e8e4d8R277 as that framework version contain the keycloak 25 and causing some problem

File: pom.xml [hunk 2]
```diff
@@ -52,7 +52,7 @@
         <exclude.quarkus.cli.tests>**/QuarkusCli*IT.java</exclude.quarkus.cli.tests>
         <exclude.quarkus.devmode.tests>no</exclude.quarkus.devmode.tests>
         <!-- Docker images used by both surefire and failsafe plugin -->
-        <postgresql.latest.image>docker.io/library/postgres:16.1</postgresql.latest.image>
+        <postgresql.latest.image>docker.io/library/postgres:16.3</postgresql.latest.image>
```
Comment (jedla97) [inline pom.xml:55]: Afaik should be better to use just `docker.io/library/postgres:16` without the minor version as it should always download the latest

**Review by jedla97 — COMMENTED** [review]
Also as upstream bumping the MySQL to 8.4 LTS shouldn't we also bump the `mysql.upstream.80.image` to `docker.io/library/mysql:8.4`

File: pom.xml (hunk 1 above)
Comment (fedinskiy) [reply pom.xml]: Rebased on that change, thank for pointing out!

File: pom.xml (hunk 2 above)
Comment (fedinskiy) [reply pom.xml:55]: Good point, changed it.

**Comment by fedinskiy** [comment]
run tests

**Comment by fedinskiy** [comment]
run tests

**Review by jedla97 — APPROVED** [review]
Looks good to me. As I was originaly wasn't ask to review, feel free to ask someone else otherwise I think we can merge it

**Pull request merged**